common/monitor.go

```diff
@@ -1,158 +1,156 @@
+package common
+
 // Copyright (c) of parts are held by the various contributors (see the CLA)
 // Licensed under the MIT License. See LICENSE file in the project root for full license information.
-package common
 
 import (
-	"github.com/FactomProject/factom"
-	"math/rand"
 	"sync"
 	"time"
+
+	"github.com/FactomProject/factom"
+	"github.com/cenkalti/backoff"
 )
 
-// FactomdMonitor
-// Running multiple Monitors is problematic and should be avoided if possible
-type FactomdMonitor struct {
-	root       bool            // True if this is the root FactomMonitor
-	mutex      sync.Mutex      // Protect multiple parties accessing monitor data
-	lastminute int64           // Last minute we got
-	lastblock  int64           // Last block we got
-	polltime   int64           // How frequently do we poll
-	kill       chan int        // Channel to kill polling.
-	response   chan int        // Respond when we have stopped
-	alerts     []chan FDStatus // Channels to send minutes to
-	polls      int64
-	info       *factom.CurrentMinuteInfo
-	status     string
+var monitor *Monitor
+var once sync.Once
+
+// GetMonitor returns the singleton instance of the Factomd Monitor
+func GetMonitor() *Monitor {
+	once.Do(func() {
+		monitor = new(Monitor)
+		monitor.errors = []chan error{}
+		monitor.listeners = []chan MonitorEvent{}
+		go monitor.poll()
+	})
+	return monitor
 }
 
-func (f *FactomdMonitor) GetAlert() chan FDStatus {
-	f.mutex.Lock()
-	defer f.mutex.Unlock()
-	alert := make(chan FDStatus, 10)
-	f.alerts = append(f.alerts, alert)
-	return alert
+// Monitor polls a factomd node and sends alerts whenever the block height or minute changes
+type Monitor struct {
+	polls   int64
```

Comment:
I know it's not your code, but I am wondering what conclusion one can draw from knowing how many time we polled? I'd trim that personally.

Reply:
agreed, removed it.